File: clients/cpp/model/JumpState.cpp
#include "JumpState.hpp"

JumpState::JumpState() { }
JumpState::JumpState(bool canJump, double speed, double maxTime, bool canCancel) : canJump(canJump), speed(speed), maxTime(maxTime), canCancel(canCancel) { }
JumpState JumpState::readFrom(InputStream& stream) {
    JumpState result;
    result.canJump = stream.readBool();
    result.speed = stream.readDouble();
    result.maxTime = stream.readDouble();
    result.canCancel = stream.readBool();
    return result;
}
void JumpState::writeTo(OutputStream& stream) const {
    stream.write(canJump);
    stream.write(speed);
    stream.write(maxTime);
    stream.write(canCancel);
}
std::string JumpState::toString() const {
    return std::string("JumpState") + "(" +
        (canJump ? "true" : "false") + 
        std::to_string(speed) +
        std::to_string(maxTime) +
        (canCancel ? "true" : "false") + 
        ")";
}

File: clients/go/model/jump_state.go
package model

import (
	"io"

	mStream "../stream"
)

type JumpState struct {
	CanJump   bool
	Speed     float64
	MaxTime   float64
	CanCancel bool
}

func NewJumpState(canJump bool, speed float64, maxTime float64, canCancel bool) *JumpState {
	return &JumpState{
		CanJump:   canJump,
		Speed:     speed,
		MaxTime:   maxTime,
		CanCancel: canCancel,
	}
}
func ReadJumpState(reader io.Reader) *JumpState {
	return &JumpState{
		CanJump:   mStream.ReadBool(reader),
		Speed:     mStream.ReadFloat64(reader),
		MaxTime:   mStream.ReadFloat64(reader),
		CanCancel: mStream.ReadBool(reader),
	}
}
func (value JumpState) Write(writer io.Writer) {
	mStream.WriteBool(writer, value.CanJump)
	mStream.WriteFloat64(writer, value.Speed)
	mStream.WriteFloat64(writer, value.MaxTime)
	mStream.WriteBool(writer, value.CanCancel)
}
